### ISSUE TITLE
Detecting java.lang.Deprecated annotation

### ISSUE THREAD

**Issue by DariusX** [CodeNarc/CodeNarc#746]
It would be nice to have SonarQube report on use of methods or classes that are annotated with java.lang.Deprecated
Since SonarQube uses CodeNarc for its Groovy rule plugin, I was wondering if the addition of such a rule was possible and desirable in Codenarc

**Comment by chrismair** [comment]
Yes, I believe checking for the @Deprecated annotation would be possible, and could be a useful new rule.

Note: It would not find @deprecated comments within Javadoc/GroovyDoc comments (unless you extend the rule to also check for that, probably needing to use less accurate regular expressions and heuristics rather than the Groovy AST).
